### PR TITLE
Always include JMXFetch logs, regardless of agent log level

### PR DESCRIPTION
**Pull request by djmitche**
### What does this PR do?

Hard-codes the seelog log level to "info" for the jmxLogger instance, since all messages from JMXFetch are represented as seelog levels info and error.

### Motivation

Setting something higher in the `log_level` config (like "warn") causes JMXFetch messages -- even those at the warn or higher level -- to be omitted because they come to seelog through JMXInfo, which logs at the info level.

JMXFetch provides its own log filtering already, so there's no need to do so at the seelog level.

### Possible Drawbacks / Trade-offs

None?

### Describe how to test/QA your changes

Run an agent with JMXFetch configured and set `log_level` to "warn", and observe that JMXFetch logging still appears.

Also observe logging is still visible from `agent jmx` in the same condition.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/jmx-fetch-logs-a5ae8bdc27d13a92.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+upgrade:
+  - |
+    List upgrade notes here, or remove this section.
```
Comment (gh123man) [inline releasenotes/notes/jmx-fetch-logs-a5ae8bdc27d13a92.yaml:11]: I think you forgot to remove these sections in the reno (maybe the change wasn't staged?)

File: pkg/config/log.go [hunk 2]
```diff
@@ -102,8 +102,11 @@ func SetupLogger(loggerName LoggerName, logLevel, logFile, syslogURI string, sys
 // if a non empty logFile is provided, it will also log to the file
 // a non empty syslogURI will enable syslog, and format them following RFC 5424 if specified
 // you can also specify to log to the console and in JSON format
-func SetupJMXLogger(loggerName LoggerName, logLevel, logFile, syslogURI string, syslogRFC, logToConsole, jsonFormat bool) error {
-	seelogLogLevel, err := validateLogLevel(logLevel)
+func SetupJMXLogger(loggerName LoggerName, logFile, syslogURI string, syslogRFC, logToConsole, jsonFormat bool) error {
+	// The JMX logger always logs at level "info", because JMXFetch does its
+	// own level filtering on and provides all messages to seelog at the info
+	// or error levels, via log.JMXInfo and log.JMXError.
+	seelogLogLevel, err := validateLogLevel("info")
```
Comment (sgnn7) [inline pkg/config/log.go:109]: I could be wrong but I think using `info`/`trace`/`debug` worked fine and it was just the higher ones (`warn`/ `error`) that had issues. With this change, we would no longer be able to use debug/trace levels which would be a problem for troubleshooting/debugging if that's the case.

File: releasenotes/notes/jmx-fetch-logs-a5ae8bdc27d13a92.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    Upgrade notes should be rare: only list known/potential breaking changes,
+    or major behaviorial changes that require user action before the upgrade.
+    Notes here must include steps that users can follow to 1. know if they're
+    affected and 2. handle the change gracefully on their end.
+features:
+  - |
+    List new features here, or remove this section.
```
Comment (sgnn7) [inline releasenotes/notes/jmx-fetch-logs-a5ae8bdc27d13a92.yaml:18]: nit: All non-relevant sections should be removed or they'll carry into the release notes

**Review by sgnn7 — COMMENTED** [review]
@djmitche Left a couple of comment on this but would be good to discuss this in the daily

File: releasenotes/notes/jmx-fetch-logs-a5ae8bdc27d13a92.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+enhancements:
+  - |
+    Logs from JMXFetch will now be included in the agent logfile, regardless
+    of the ``log_level`` setting of the agent.
```
Comment (kayayarai) [inline releasenotes/notes/jmx-fetch-logs-a5ae8bdc27d13a92.yaml:22]:
```suggestion
    Logs from JMXFetch will now be included in the Agent logfile, regardless
    of the `log_level` setting of the Agent.
```

**Review by kayayarai — APPROVED** [review]
What the others mentioned, plus capitalize Agent.

File: releasenotes/notes/jmx-fetch-logs-a5ae8bdc27d13a92.yaml (hunk 4 above)
Comment (djmitche) [reply releasenotes/notes/jmx-fetch-logs-a5ae8bdc27d13a92.yaml:22]: @kayayarai this is a rST file, so double-backticks is correct.

We get this correction on _every_ PR that has a changelog -- how could we help your team save the time of making that suggestion?

Comment (kayayarai) [reply releasenotes/notes/jmx-fetch-logs-a5ae8bdc27d13a92.yaml]: @djmitche Thank you for the info, and apologies for the repeated error. I'll mention it to the team, and add it to our reviewing guidelines! Hopefully that will reduce our over-eagerness.

File: pkg/config/log.go (hunk 2 above)
Comment (sgnn7) [reply pkg/config/log.go:109]: Update from offline sync: The jmxfetch configuration which I was referring to is different from the agent's log level of processing of received messages from jmxfetch so my earlier comment shouldn't apply.

**Review by sgnn7 — APPROVED** [review]

**Pull request merged**